File: 2216-delete-the-middle-node-of-a-linked-list/2216-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head.next == None:
            return None
        prev = None
        slow = head
        quick = head
        while quick and quick.next:
            prev = slow
            slow = slow.next
            quick = quick.next.next
        prev.next = slow.next
        return head

File: 2216-delete-the-middle-node-of-a-linked-list/2216-delete-the-middle-node-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteMiddle = function(head) {
    let prev = null;
    let slow = head;
    let quick = head;
    while (quick && quick.next !== null) {
        prev = slow
        slow = slow.next;
        quick = quick.next.next
    }
    if (prev) {
        prev.next = slow.next;
        return head;   
    } else {
        return null;
    }
};